mypy/typestate.py

```diff
@@ -29,6 +33,31 @@ class TypeState:
     _subtype_caches = {}  # type: ClassVar[Dict[TypeInfo, Set[Tuple[Instance, Instance]]]]
     _subtype_caches_proper = {}  # type: ClassVar[Dict[TypeInfo, Set[Tuple[Instance, Instance]]]]
 
+    # This contains protocol dependencies generated after running a full build,
+    # or after an update. These dependencies are special because:
+    #   * They are a global property of the program; i.e. some dependencies for imported
+    #     classes can be generated in the importing modules.
+    #   * Because of the above, they are serialized separately, after a full run,
+    #     or a full update.
+    # `proto_deps` can be None if after deserialization it turns out that they are
+    # inconsistent with the other cache files (or an error occurred during deserialization).
+    # A blocking error will be generated in this case, since we can't proceed safely.
```

Comment:
Document what kinds of dependencies are stored here (of the 3 kinds discussed in the PR message) and give examples. Preferably have real code fragments, and explain exactly which dependencies would be generated and why.

Potential comment update: Can they also be `None` if we aren't using fine-grained incremental checking?

Reply:
> Can they also be `None` if we aren't using fine-grained incremental checking?

Currently they should be `None` only in case of error, otherwise it is an empty dict.